ci(lint): add trunk linting setup

rriski opened the pull request:
Best reviewed commit by commit.

* run linting with [trunk-io](https://docs.trunk.io/check)
* add markdownlint check
* add codespell check
* add yamllint check
* add hadolint check

Most of the diff is autofixed.

Serpentiel commented on .github/workflows/lint.yml:
```diff
@@ -9,25 +9,17 @@ on:
 jobs:
-  commitlint:
+  trunk:
+    if: >
```
maybe we can drop this `if` check for now, we don't really have the `skip workflows` label at the moment, if this is needed, we can always add it in future

Serpentiel commented:
otherwise lgtm

Serpentiel commented:
also please consider a different naming for the first commit/PR, `feat` makes sense in new product features most of the time, there are other good tags that work here, e.g. `ci`

rriski commented:
> also please consider a different naming for the first commit/PR, `feat` makes sense in new product features most of the time, there are other good tags that work here, e.g. `ci`, `build`, etc.

Fixed the first commit message and removed `if` from github action

Serpentiel commented on .github/workflows/lint.yml:
```diff
@@ -9,25 +9,14 @@ on:
       - main
 
 jobs:
-  commitlint:
```
let's keep `commitlint`: sadly Trunk doesn't/can't have this functionality, because it has to be executed on a higher level than Trunk operates on, i.e. it should be GitHub-aware, and it's not

Serpentiel reviewed (changes requested):
lgtm but cant merge because of `commitlint`

Serpentiel reviewed (approved):
lgtm
